packages/plugin-ext/src/main/browser/menus/menus-contribution-handler.ts

```diff
@@ -266,36 +266,50 @@ export class MenusContributionPointHandler {
             const uri = this.resourceContextKey.get();
             return uri ? uri['codeUri'] : undefined;
         };
-        this.registerMenuAction(menuPath, menu, {
-            execute: () => this.commands.executeCommand(menu.command, selectedResource()),
-            isEnabled: () => this.commands.isEnabled(menu.command, selectedResource()),
-            isVisible: () => this.commands.isVisible(menu.command, selectedResource())
-        });
+        this.registerMenuAction(menuPath, menu, command => ({
+            execute: () => this.commands.executeCommand(command, selectedResource()),
+            isEnabled: () => this.commands.isEnabled(command, selectedResource()),
+            isVisible: () => this.commands.isVisible(command, selectedResource())
+        }));
     }
 
-    protected registerMenuAction(menuPath: MenuPath, menu: Menu, handler: CommandHandler): void {
-        const commandId = this.createSyntheticCommandId(menu, { prefix: '__plugin.menu.action.' });
+    protected registerMenuAction(menuPath: MenuPath, menu: Menu, handler: (command: string) => CommandHandler): void {
+        const commandId = this.createSyntheticCommandId(menu.command, { prefix: '__plugin.menu.action.' });
         const command: Command = { id: commandId };
-        this.commands.registerCommand(command, handler);
+        this.commands.registerCommand(command, handler(menu.command));
+
+        const altId = menu.alt && this.createSyntheticCommandId(menu.alt, { prefix: '__plugin.menu.action.' });
+        const alt: Command | undefined = altId && { id: altId } || undefined;
+        if (alt) {
+            this.commands.registerCommand(alt, handler(menu.alt!));
```

Comment:
Would this be the same as:
```js
this.commands.registerCommand(alt, handler(alt.id));
```
And you could omit the `!`.

Reply:
Oh, I see; no. We need the original, alternative command ID instead of the prefixed one, right?

Reply:
`alt.id` is a synthetic command, we pass here the real command id `menu.alt`

I will need to add `menu.alt` check to `if` to get rid of `!`, but it is not necessary since `alt` can be defined only then `menu.alt` is defined.